mountpoint-s3/src/inode.rs

```diff
@@ -548,6 +557,29 @@ impl SuperblockInner {
     }
 }
 
+/// Data from a remote object.
+#[derive(Debug)]
+pub struct RemoteLookup {
+    kind: InodeKind,
+    stat: InodeStat,
+}
+
+/// Result of a call to [SuperblockInner::try_update_child].
+#[derive(Debug)]
+enum UpdateStatus<'n, 'r> {
```

Comment:
The lifetime for this struct is a bit complicated and I noticed that we still have to clone the data when using. Can we simplify this by taking ownership of these variable?